### ISSUE TITLE
iOS9: topLayoutGuide/bottomLayoutGuide cause crash

### ISSUE THREAD

**Issue by zhudongfang** [SnapKit/Masonry#242]
Code in demo
```
[topView makeConstraints:^(MASConstraintMaker *make) {
    UIView *topLayoutGuide = (id)self.topLayoutGuide;
    // topLayoutGuide cause exception
    make.top.equalTo(topLayoutGuide.mas_bottom);
    make.left.equalTo(self.view);
    make.right.equalTo(self.view);
    make.height.equalTo(@40);
}];
```

**Comment by robertjpayne** [comment]
@zhudongfang mind posting the exception that it is throwing? Typically you have to do `self.topLayoutGuide.bottom` or `self.topLayoutGuide.top` which I believe is why.

**Comment by zhudongfang** [comment]
![Uploading Bitmap.png…]()

**Comment by zhudongfang** [comment]
@robertjpayne 
![bitmap](https://cloud.githubusercontent.com/assets/3676676/9951803/4a5a30ba-5dfe-11e5-9d99-4ce09c3cdb33.png)

**Comment by robertjpayne** [comment]
@zhudongfang I think iOS 9 has breaking changes so typecasting it to a UIView is no longer valid. But I'm not entirely sure how to allow it to work for iOS 8 and iOS 9.

**Comment by donpironet** [comment]
Got the same problem. Hope it will be fixed fast.

**Comment by llb1119** [comment]
The updating of Masonry's version is too slowly

**Comment by donpironet** [comment]
You mean that it will take a while until the library is up to date?

**Comment by robertjpayne** [comment]
Hey all, there is no documentation for Masonry that states you are supposed to chain the layout guides. It's a bit weird that worked before but anyways I've pushed an update that includes new API's for the layout guides:

`topLayoutGuideTop / topLayoutGuideBottom / bottomLayoutGuideTop / bottomLayoutGuideBottom` the existing api's remain `topLayoutGuide` is identical to `topLayoutGuideBottom` and `bottomLayoutGuide` is identical to `bottomLayoutGuideTop` since they are the most common use cases.

So in summary, do not attempt to use `mas_*` from a layout guide, use them as is and they should work fine.

This is also available now via v0.6.3

**Comment by donpironet** [comment]
when is this going to be on cocoapods? Thanks for you fast work btw.

**Comment by robertjpayne** [comment]
@cloudkite just needs to do the push to cocoapods.

**Comment by cloudkite** [comment]
cocoapods is now at 0.6.3 Thanks @robertjpayne for the fast fix

**Comment by donpironet** [comment]
@robertjpayne For me it's not clear how to use the new API now. I thought it was something like:

make.top.equalTo(self.topLayoutGuide.topLayoutGuideBottom);

But that doesn't compile, so could you please give an example?

**Comment by warpling** [comment]
@robertjpayne thank you so much for rushing this out!!
@donpironet it looks like the shortcuts aren't available so you'll have to use 
````
make.top.equalTo(self.mas_topLayoutGuide);
````

**Comment by napolux** [comment]
This was very helpful to solve a mysterious crash I was seeing. Thanks! 👍
